ClientRuntimes/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -64,74 +63,52 @@ def get_put_operation_result(azure_response, custom_deserialization_block)
         fail polling_error if polling_error.is_a?(Exception)
       end
 
-      if (AsyncOperationStatus.is_successful_status(polling_state.status) && polling_state.resource.nil?)
+      if (http_method === :put || http_method === :patch) && AsyncOperationStatus.is_successful_status(polling_state.status) && polling_state.resource.nil?
         get_request = MsRest::HttpOperationRequest.new(request.base_uri, request.build_path.to_s, 'get', query_params: request.query_params)
         update_state_from_get_resource_operation(get_request, polling_state, custom_deserialization_block)
       end
 
-      if (AsyncOperationStatus.is_failed_status(polling_state.status))
+      if AsyncOperationStatus.is_failed_status(polling_state.status)
         fail polling_state.get_operation_error
       end
 
-      return polling_state.get_operation_response
+      polling_state.get_operation_response
+    end
+
+    # @TODO Update name of the method to get_put_or_patch_operation_result when introducing breaking change / updating major version of gem
+    # Retrieves the result of 'PUT' or 'PATCH' operation. Performs polling of required.
+    # @param azure_response [MsRestAzure::AzureOperationResponse] response from Azure service.
+    # @param custom_deserialization_block [Proc] custom logic for response deserialization.
+    #
+    # @return [MsRest::HttpOperationResponse] the response.
+    def get_put_operation_result(azure_response, custom_deserialization_block)
```

Comment:
Let's check on whether we need to keep this one or not per our offline conversation.